executor/insert_test.go

```diff
@@ -287,6 +287,231 @@ func (s *testSuite3) TestAllowInvalidDates(c *C) {
+	tk.MustExec(`create table t6(id int primary key, n double auto_increment, key I_n(n));`)
+	tk.MustExec(`create table t7(id int primary key, n double unsigned auto_increment, key I_n(n));`)
+
+	tests := []struct {
+		insert string
+		query  string
+		result [][]interface{}
+	}{
+		{
+			`insert into t1(id, n) values(1, 1)`,
+			`select * from t1 where id = 1`,
+			testkit.Rows(`1 1`),
+		},
+		{
+			`insert into t1(n) values(2)`,
+			`select * from t1 where id = 2`,
+			testkit.Rows(`2 2`),
+		},
+		{
+			`insert into t1(n) values(3)`,
+			`select * from t1 where id = 3`,
+			testkit.Rows(`3 3`),
+		},
+		{
+			`insert into t1(id, n) values(-1, 4)`,
+			`select * from t1 where id = -1`,
+			testkit.Rows(`-1 4`),
+		},
+		{
+			`insert into t1(n) values(5)`,
+			`select * from t1 where id = 4`,
+			testkit.Rows(`4 5`),
+		},
+		{
+			`insert into t1(id, n) values('5', 6)`,
+			`select * from t1 where id = 5`,
+			testkit.Rows(`5 6`),
+		},
+		{
+			`insert into t1(n) values(7)`,
+			`select * from t1 where id = 6`,
+			testkit.Rows(`6 7`),
+		},
+		{
+			`insert into t1(id, n) values(7.4, 8)`,
+			`select * from t1 where id = 7`,
+			testkit.Rows(`7 8`),
+		},
+		{
+			`insert into t1(id, n) values(7.5, 9)`,
+			`select * from t1 where id = 8`,
+			testkit.Rows(`8 9`),
+		},
+		{
+			`insert into t1(n) values(9)`,
+			`select * from t1 where id = 8`,
+			testkit.Rows(`8 9`),
+		},
+		{
+			`insert into t2(id, n) values(1, 1)`,
+			`select * from t2 where id = 1`,
+			testkit.Rows(`1 1`),
+		},
+		{
+			`insert into t2(n) values(2)`,
+			`select * from t2 where id = 2`,
+			testkit.Rows(`2 2`),
+		},
+		{
+			`insert into t2(n) values(3)`,
+			`select * from t2 where id = 3`,
+			testkit.Rows(`3 3`),
+		},
+		{
+			`insert into t3(id, n) values(1, 1)`,
+			`select * from t3 where id = 1`,
+			testkit.Rows(`1 1`),
+		},
+		{
+			`insert into t3(n) values(2)`,
+			`select * from t3 where id = 2`,
+			testkit.Rows(`2 2`),
+		},
+		{
+			`insert into t3(n) values(3)`,
+			`select * from t3 where id = 3`,
+			testkit.Rows(`3 3`),
+		},
+		{
+			`insert into t3(id, n) values(-1, 4)`,
+			`select * from t3 where id = -1`,
+			testkit.Rows(`-1 4`),
+		},
+		{
+			`insert into t3(n) values(5)`,
+			`select * from t3 where id = 4`,
+			testkit.Rows(`4 5`),
+		},
+		{
+			`insert into t4(id, n) values(1, 1)`,
+			`select * from t4 where id = 1`,
+			testkit.Rows(`1 1`),
+		},
+		{
+			`insert into t4(id) values(2)`,
+			`select * from t4 where id = 2`,
+			testkit.Rows(`2 2`),
+		},
+		{
+			`insert into t4(id, n) values(3, -1)`,
+			`select * from t4 where id = 3`,
+			testkit.Rows(`3 -1`),
+		},
+		{
+			`insert into t4(id) values(4)`,
+			`select * from t4 where id = 4`,
+			testkit.Rows(`4 3`),
+		},
+		{
+			`insert into t4(id, n) values(5, 5.5)`,
+			`select * from t4 where id = 5`,
+			testkit.Rows(`5 5.5`),
+		},
+		{
+			`insert into t4(id) values(6)`,
+			`select * from t4 where id = 6`,
+			testkit.Rows(`6 6`),
```

Comment:
It isn't compatible with MySQL5.7, the result of MySQL is as follows:
```
mysql> select * from t4 where id = 6;
+----+---+
| id | n |
+----+---+
|  6 | 7 |
+----+---+
1 row in set (0.01 sec)
```

Reply:
@zimulala fixed! thank you! 
BTW, the update behavior is the same as MySQL5.8, but is not the same as MySQL5.7.  I think the behavior of MySQL5.8 is expected to me.